api/test/trace/global_provider_test.cc

```diff
@@ -0,0 +1,34 @@
+#include "opentelemetry/trace/global_provider.h"
+
+#include <gtest/gtest.h>
+
+using opentelemetry::trace::Provider;
+using opentelemetry::trace::Tracer;
+
+class TestProvider : public opentelemetry::trace::TracerProvider
+{
+  Tracer *const GetTracer(opentelemetry::nostd::string_view library_name,
+                          opentelemetry::nostd::string_view library_version,
+                          const opentelemetry::trace::TracerOptions &tracer_options) override
+  {
+    return nullptr;
+  }
+};
+
+TEST(Provider, GetTracerProviderDefault)
+{
+  auto tf = Provider::GetTracerProvider();
+  ASSERT_NE(tf, nullptr);
```

Comment:
1. (optional) this can be EXPECT instead of ASSERT.
2. (nit) The order is EXPECT_NE(expected, actual);

(except when it comes to EXPECT_THAT where it's the opposite: EXPECT_THAT(actual, EqualsExpected(...)))

Reply:
Haha, good to know! Thanks for the heads up.